ePub3/ePub/container.cpp

```diff
@@ -300,9 +300,15 @@ void Container::LoadEncryption()
 }
 shared_ptr<EncryptionInfo> Container::EncryptionInfoForPath(const string &path) const
 {
+    // Compare encryption path with URI-encoded and decoded paths
+    url_canon::RawCanonOutputW<256> output;
+    url_util::DecodeURLEscapeSequences(path.c_str(), static_cast<int>(path.utf8_size()), &output);
+    string decodedPath(output.data(), output.length());
+
     for ( auto item : _encryption )
     {
-        if ( item->Path() == path )
+        string itemPath = item->Path();
+        if ( itemPath == path || itemPath == decodedPath )
```

Comment:
better safe than sorry and check both cases, but I was wondering since we go to the effort of url-decoding, isn't the first check redundant? Or are there cases where the encrypted file contains the URL encoded version of the path?

Reply:
without statistics of the use of encoded/decoded URIs in `encryption.xml`, removing the first check would be a bit too brave... technically, nothing prevents the use of URI-encoded symbols in the file.

Reply:
sounds good, yeah mine was just a curiosity to see if we knew more about its usage.